src/Symfony/Component/Config/Definition/Builder/NodeBuilder.php

```diff
@@ -133,7 +133,7 @@ public function variableNode($name)
     /**
      * Returns the parent node.
      *
-     * @return ParentNodeDefinitionInterface|NodeDefinition The parent node
+     * @return NodeDefinition&ParentNodeDefinitionInterface The parent node
```

Comment:
I don't think we have any occurrence of this syntax in the codebase.
Can you check if it's compatible with usual tooling, eg phpstorm, phpdocumentor, etc?

Reply:
[Next version of PHPStorm](https://blog.jetbrains.com/phpstorm/2018/09/phpstorm-2018-3-eap-183-2635-12/) will support it (and falls back to the first type currently); there's a well-timed [proposal for PSR-5](https://github.com/php-fig/fig-standards/pull/1104); untested but [phpDocumentor probably doesn't work](https://github.com/vimeo/psalm/issues/594#issuecomment-373943107).

This syntax was in https://wiki.php.net/rfc/intersection_types too.